### PR TITLE
feat(api): implement fetch by semver

### PR DESCRIPTION
**Pull request by alexbrdn**
- feat(api): implement fetch by semver
- fix(fetch): match partial semantic version correctly
- feat: remove possibility to fetch by version digest

### CODE REVIEW

File: cmd/fetch.go [hunk 1]
```diff
@@ -10,11 +10,12 @@ import (
...
 
 var fetchCmd = &cobra.Command{
-	Use:   "fetch (<NAME[:SEMVER|DIGEST]> | <TMID>)",
-	Short: "Fetches the TM by name or id",
-	Long:  `Fetches TM by name, optionally accepting semantic version or digest.`,
-	Args:  cobra.ExactArgs(1),
-	Run:   executeFetch,
+	Use:   "fetch <NAME>[:<SEMVER>] | <TMID>)",
+	Short: "Fetches a TM by name or id",
+	Long: `Fetches a TM by name, optionally accepting a semantic version.
```
Comment (enricoschuetz) [inline cmd/fetch.go:15]: --help shows the long text. 
In the current long text it is not mentioned that you can also fetch by full ID.

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,6 +7,7 @@
 - Implemented file locking to avoid data races (guaranteed for Linux/MacOSX)
 - Implemented ```pull``` command to fetch and save multiple thing models at once
 - Implemented setting CORS options for API
+- Implemented fetching a TM by a \[partial\] semantic version or version digest also in REST API
```
Comment (enricoschuetz) [inline CHANGELOG.md:10]: By "digest" should be removed from the comment, as it is not supported anymore.

Comment (alexbrdn) [reply CHANGELOG.md]: removed

File: cmd/fetch.go (hunk 1 above)
Comment (alexbrdn) [reply cmd/fetch.go]: good point. fixed

**Review by enricoschuetz — APPROVED** [review]

**Pull request merged**